pkg/s3wrapper/filesystem.go

```diff
@@ -117,17 +118,26 @@ func (f *FSClient) UploadStream(ctx context.Context, reader io.Reader, objectNam
 		return err
 	}
 	buffer := make([]byte, 1024)
-	fo, err := os.Create(filePath)
+
+	t, err := renameio.TempFile("", filePath)
 	if err != nil {
-		err = errors.Wrapf(err, "Unable to open file for writing %s", filePath)
+		err = errors.Wrapf(err, "Unable to create a temp file for %s", filePath)
+		log.Error(err)
+		return err
+	}
+
+	if err := t.Chmod(0600); err != nil {
```

Comment:
I don't think you need this. From the `renameio` docs:
> The file's permissions will be 0600 by default

ref: https://pkg.go.dev/github.com/google/renameio#TempFile

Reply:
Thanks, removed.